### PR TITLE
docs(README.md, CONTRIBUTING.md): Update and Add respectively

### PR DESCRIPTION
**Pull request by knittingcodemonkey**
- Update README.md to be more descriptive. 
- Add CONTRIBUTING.md along the same guidelines as RxJs CONTRIBUTING.md
- Add commitizen, cz-conventional-changelog to dev dependencies
- Add commizen config to package.json to reference cz-conventional-changelog
- Add commit script to package.json to run commitizen as is used by the RxJs repo

No breaking changes. Assists in closing https://github.com/ReactiveX/rxjs-docs/issues/19

### CODE REVIEW

**Comment by ladyleet** [comment]
LGTM @knittingcodemonkey! thanks for your 2nd contribution! @kwonoj 👍?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+* Create your patch, following [code style guidelines](#coding-style-guidelines), and **including appropriate test cases**.
+* Run the full test suite and ensure that all tests pass.
+* Run the micro and macro performance tests against your feature branch and compare against master
+  to ensure performance wasn't changed for the worse.
+* Commit your changes using a descriptive commit message that follows our
+  [commit message guidelines](#commit-message-guidelines). Adherence to these conventions
+  is necessary because release notes are automatically generated from these messages.
+
+     ```shell
+     git commit -a
```
Comment (kwonoj) [inline CONTRIBUTING.md]: not for this PR, but probably we should have same tooling as rxjs repo to support `npm run commit` commit wizard.

Comment (kwonoj) [reply CONTRIBUTING.md]: welp, actually this PR has it - let's add into guides as well (and recommend to use it)

**Comment by kwonoj** [comment]
Super nitty comment otherwise lgtm. I'd recommdnd ppl use `npm run commit` daily instead of direct commit.

**Comment by ladyleet** [comment]
Felt so good to merge this! Thx @knittingcodemonkey!

**Pull request merged**